src/plugins/vis_type_vega/public/vega_view/vega_map_view/view.ts

```diff
@@ -0,0 +1,185 @@
+      zoomSettings.maxZoom = (await tmsService.getMaxZoom()) ?? defaultMapConfig.maxZoom;
+      zoomSettings.minZoom = (await tmsService.getMinZoom()) ?? defaultMapConfig.minZoom;
+      customAttribution = await this.mapServiceSettings.getAttributionsForTmsService(tmsService);
+      style = (await tmsService.getVectorStyleSheet()) as Style;
+    } else {
+      customAttribution = this.mapServiceSettings.config.tilemap.options.attribution;
+    }
+
+    // In some cases, Vega may be initialized twice, e.g. after awaiting...
+    if (!this._$container) return;
+
+    const mapBoxInstance = new Map({
+      style,
+      customAttribution,
+      container: this._$container.get(0),
+      ...this.getMapParams({ ...zoomSettings }),
+    });
+
+    const initMapComponents = () => {
+      this.initControls(mapBoxInstance);
+      this.initLayers(mapBoxInstance, vegaView);
+
+      this._addDestroyHandler(() => {
+        if (mapBoxInstance.getLayer(vegaLayerId)) {
+          mapBoxInstance.removeLayer(vegaLayerId);
+        }
+        if (mapBoxInstance.getLayer(userConfiguredLayerId)) {
+          mapBoxInstance.removeLayer(userConfiguredLayerId);
+        }
+        mapBoxInstance.remove();
+      });
+    };
+
+    mapBoxInstance.once('load', initMapComponents);
+  }
+
+  private async updateVegaView(mapBoxInstance: Map, vegaView: vega.View) {
+    const mapCanvas = mapBoxInstance.getCanvas();
+    const { lat, lng } = mapBoxInstance.getCenter();
+    let shouldRender = false;
+
+    const sendSignal = (sig: string, value: any) => {
+      if (vegaView.signal(sig) !== value) {
+        vegaView.signal(sig, value);
+        shouldRender = true;
+      }
+    };
+
+    sendSignal('width', mapCanvas.clientWidth);
+    sendSignal('height', mapCanvas.clientHeight);
+    sendSignal('latitude', lat);
+    sendSignal('longitude', lng);
+    sendSignal('zoom', mapBoxInstance.getZoom());
+
+    if (shouldRender) {
+      await vegaView.runAsync();
+    }
+  }
+
+  private initControls(mapBoxInstance: Map) {
+    if (this.shouldShowZoomControl) {
+      mapBoxInstance.addControl(new NavigationControl({ showCompass: false }), 'top-left');
+    }
+  }
+
+  private initLayers(mapBoxInstance: Map, vegaView: vega.View) {
+    const shouldShowUserConfiguredLayer = this.mapStyle === TmsTileLayers.userConfigured;
+
+    if (shouldShowUserConfiguredLayer) {
+      const { url, options } = this.mapServiceSettings.config.tilemap;
+
+      initTmsRasterLayer({
+        id: userConfiguredLayerId,
+        map: mapBoxInstance,
+        context: {
+          tiles: [url!],
+          maxZoom: options.maxZoom ?? defaultMapConfig.maxZoom,
+          minZoom: options.minZoom ?? defaultMapConfig.minZoom,
+          tileSize: options.tileSize ?? defaultMapConfig.tileSize,
+        },
+      });
+    }
+
+    initVegaLayer({
+      id: vegaLayerId,
+      map: mapBoxInstance,
+      context: {
+        vegaView,
+        updateVegaView: throttle(this.updateVegaView, 8),
```

Comment:
where does the `8` ms come from?

I don't think we need it.

`Layer#render`

https://github.com/elastic/kibana/blob/216f13064709f6d044bbc27988722bca0a3c1844/src/plugins/vis_type_vega/public/vega_view/vega_map_view/layers/vega_layer.ts#L41

is already throttled to the framerate of the device. see https://docs.mapbox.com/mapbox-gl-js/api/properties/#customlayerinterface#render

So I'd just do:

```
  updateVegaView: (mapboxInstance, vegaView) => {
            this.updateVegaView(mapboxInstane, vegaView);
};